Change pydusa recipe

mstabrin opened the pull request:
We figured out, that the pydusa recipe is always using the latest master branch and this resulted in a broken installer a couple of days ago.

I added version tags to the pydusa repository and changed the recipe/installation script, so that users will use the default version (in this case tag v20170831), but developers do still have the freedom to build other tags/branches by adding a second argument next to the numpy version.

@shadowwalkersb Do you agree with the changes or do you have another way of doing it?
I am not really familiar with anaconda recipes yet.

sludtke42 commented:
Hi Markus,
as you may be aware we have been going through a significant weather event the last few days. Tunay is fine, but he was fairly close to being flooded, and none of us got much done for the last 5 days or so. I expect he will be coming back online shortly. Things are recovering well now. Please be patient.

> On Aug 31, 2017, at 8:38 AM, Markus Stabrin <notifications@github.com> wrote:
> 
> We figured out, that the pydusa recipe is always using the latest master branch and this resulted in a broken installer a couple of days ago.
> 
> I added version tags to the pydusa repository and changed the recipe/installation script, so that users will use the default version (in this case tag v20170831), but developers do still have the freedom to build other tags/branches by adding a second argument next to the numpy version.
> 
> @shadowwalkersb <https://github.com/shadowwalkersb> Do you agree with the changes or do you have another way of doing it?
> I am not really familiar with anaconda recipes yet.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/cryoem/eman2/pull/119 <https://github.com/cryoem/eman2/pull/119>
> Commit Summary
> 
> Changed pydusa recipe to make other versions available
> Removed source activate root again
> File Changes
> 
> M recipes/pydusa/meta.yaml <https://github.com/cryoem/eman2/pull/119/files#diff-0> (4)
> M utils/build_pydusa_numpy.sh <https://github.com/cryoem/eman2/pull/119/files#diff-1> (9)
> Patch Links:
> 
> https://github.com/cryoem/eman2/pull/119.patch <https://github.com/cryoem/eman2/pull/119.patch>
> https://github.com/cryoem/eman2/pull/119.diff <https://github.com/cryoem/eman2/pull/119.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6vQsxJ5iIdUMo-hxNx1mz-kGlOtTks5sdrdfgaJpZM4PI14s>.
> 

----------------------------------------------------------------------------
Steven Ludtke, Ph.D.
Professor, Dept of Biochemistry and Mol. Biol.         (www.bcm.edu/biochem)
Co-Director National Center For Macromolecular Imaging        (ncmi.bcm.edu)
Co-Director CIBR Center                          (www.bcm.edu/research/cibr)
Baylor College of Medicine                             
sludtke@bcm.edu

mstabrin commented:
Hello Steve,

Glad to hear that things are recovering well and that you all are fine and thank you for the status update.

As the news all over the world are full of Houston updates (and the pictures are still horrible) I, of course, did not expect a fast answer.

Best wishes to all of you!
Markus

On 31 Aug 2017, at 16:21, Steve Ludtke <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Markus,
as you may be aware we have been going through a significant weather event the last few days. Tunay is fine, but he was fairly close to being flooded, and none of us got much done for the last 5 days or so. I expect he will be coming back online shortly. Things are recovering well now. Please be patient.

> On Aug 31, 2017, at 8:38 AM, Markus Stabrin <notifications@github.com<mailto:notifications@github.com>> wrote:
>
> We figured out, that the pydusa recipe is always using the latest master branch and this resulted in a broken installer a couple of days ago.
>
> I added version tags to the pydusa repository and changed the recipe/installation script, so that users will use the default version (in this case tag v20170831), but developers do still have the freedom to build other tags/branches by adding a second argument next to the numpy version.
>
> @shadowwalkersb <https://github.com/shadowwalkersb> Do you agree with the changes or do you have another way of doing it?
> I am not really familiar with anaconda recipes yet.
>
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/cryoem/eman2/pull/119 <https://github.com/cryoem/eman2/pull/119>
> Commit Summary
>
> Changed pydusa recipe to make other versions available
> Removed source activate root again
> File Changes
>
> M recipes/pydusa/meta.yaml <https://github.com/cryoem/eman2/pull/119/files#diff-0> (4)
> M utils/build_pydusa_numpy.sh <https://github.com/cryoem/eman2/pull/119/files#diff-1> (9)
> Patch Links:
>
> https://github.com/cryoem/eman2/pull/119.patch <https://github.com/cryoem/eman2/pull/119.patch>
> https://github.com/cryoem/eman2/pull/119.diff <https://github.com/cryoem/eman2/pull/119.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6vQsxJ5iIdUMo-hxNx1mz-kGlOtTks5sdrdfgaJpZM4PI14s>.
>

----------------------------------------------------------------------------
Steven Ludtke, Ph.D.
Professor, Dept of Biochemistry and Mol. Biol. (www.bcm.edu/biochem<http://www.bcm.edu/biochem>)
Co-Director National Center For Macromolecular Imaging (ncmi.bcm.edu<http://ncmi.bcm.edu>)
Co-Director CIBR Center (www.bcm.edu/research/cibr<http://www.bcm.edu/research/cibr>)
Baylor College of Medicine
sludtke@bcm.edu<mailto:sludtke@bcm.edu>






—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/cryoem/eman2/pull/119#issuecomment-326310676>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ARNjSBsXuOpr5pG2cOydc9Ku64IUR-Sgks5sdsFrgaJpZM4PI14s>.

sludtke42 commented:
Not quite as bad as the news is making it sound. They always hunt out the worst possible locations then stress them. Estimates are about 50,000 people displaced out of a population of 6,000,000, maybe as many as 200,000 with some significant impact. Still, this is a relatively small percentage of the population, so dealing with it is feasible (a lot of people available to help each person who suffered). Financial impact will be massive. Most of the flooding occured in areas known to be at risk of flooding, though extending further than usual. There were a lot of people who were worried that they might be flooded during the worst of the rain, but wound up ok. We had almost exactly 1 m (yes, meter) of rain at our house, and the worst the flooding got was 6-8" in the street.  Houston is well designed for floods!


> On Aug 31, 2017, at 9:34 AM, Markus Stabrin <notifications@github.com> wrote:
> 
> Hello Steve,
> 
> Glad to hear that things are recovering well and that you all are fine and thank you for the status update.
> 
> As the news all over the world are full of Houston updates (and the pictures are still horrible) I, of course, did not expect a fast answer.
> 
> Best wishes to all of you!
> Markus
> 
> On 31 Aug 2017, at 16:21, Steve Ludtke <notifications@github.com<mailto:notifications@github.com>> wrote:
> 
> Hi Markus,
> as you may be aware we have been going through a significant weather event the last few days. Tunay is fine, but he was fairly close to being flooded, and none of us got much done for the last 5 days or so. I expect he will be coming back online shortly. Things are recovering well now. Please be patient.
> 
> > On Aug 31, 2017, at 8:38 AM, Markus Stabrin <notifications@github.com<mailto:notifications@github.com>> wrote:
> >
> > We figured out, that the pydusa recipe is always using the latest master branch and this resulted in a broken installer a couple of days ago.
> >
> > I added version tags to the pydusa repository and changed the recipe/installation script, so that users will use the default version (in this case tag v20170831), but developers do still have the freedom to build other tags/branches by adding a second argument next to the numpy version.
> >
> > @shadowwalkersb <https://github.com/shadowwalkersb> Do you agree with the changes or do you have another way of doing it?
> > I am not really familiar with anaconda recipes yet.
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/cryoem/eman2/pull/119 <https://github.com/cryoem/eman2/pull/119>
> > Commit Summary
> >
> > Changed pydusa recipe to make other versions available
> > Removed source activate root again
> > File Changes
> >
> > M recipes/pydusa/meta.yaml <https://github.com/cryoem/eman2/pull/119/files#diff-0> (4)
> > M utils/build_pydusa_numpy.sh <https://github.com/cryoem/eman2/pull/119/files#diff-1> (9)
> > Patch Links:
> >
> > https://github.com/cryoem/eman2/pull/119.patch <https://github.com/cryoem/eman2/pull/119.patch>
> > https://github.com/cryoem/eman2/pull/119.diff <https://github.com/cryoem/eman2/pull/119.diff>
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6vQsxJ5iIdUMo-hxNx1mz-kGlOtTks5sdrdfgaJpZM4PI14s>.
> >
> 
> ----------------------------------------------------------------------------
> Steven Ludtke, Ph.D.
> Professor, Dept of Biochemistry and Mol. Biol. (www.bcm.edu/biochem<http://www.bcm.edu/biochem>)
> Co-Director National Center For Macromolecular Imaging (ncmi.bcm.edu<http://ncmi.bcm.edu>)
> Co-Director CIBR Center (www.bcm.edu/research/cibr<http://www.bcm.edu/research/cibr>)
> Baylor College of Medicine
> sludtke@bcm.edu<mailto:sludtke@bcm.edu>
> 
> 
> 
> 
> 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<https://github.com/cryoem/eman2/pull/119#issuecomment-326310676>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ARNjSBsXuOpr5pG2cOydc9Ku64IUR-Sgks5sdsFrgaJpZM4PI14s>.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119#issuecomment-326314790>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6l4SMEXxUZqdzbyA6Fgqu6jtnib_ks5sdsR3gaJpZM4PI14s>.
> 

----------------------------------------------------------------------------
Steven Ludtke, Ph.D.
Professor, Dept of Biochemistry and Mol. Biol.         (www.bcm.edu/biochem)
Co-Director National Center For Macromolecular Imaging        (ncmi.bcm.edu)
Co-Director CIBR Center                          (www.bcm.edu/research/cibr)
Baylor College of Medicine                             
sludtke@bcm.edu

pawelpap9 commented:
Hi

The first line of defense in the US are always volunteers. It is extremely impressive, people use their own trucks and boats and drive into Houston to rescue people and their pets. There is also very strong support through local churches, many serving as shelters for those who had to evacuate. 

If you are in a position to help even small money donations would make a difference for those who lost everything. I cannot advocate any particular charity but if you google Houston flood Harvey donations you will find possibilities. Please use reputable organizations you are familiar with.

Regards,
Pawel

> On Aug 31, 2017, at 11:56 AM, Steve Ludtke <notifications@github.com> wrote:
> 
> Not quite as bad as the news is making it sound. They always hunt out the worst possible locations then stress them. Estimates are about 50,000 people displaced out of a population of 6,000,000, maybe as many as 200,000 with some significant impact. Still, this is a relatively small percentage of the population, so dealing with it is feasible (a lot of people available to help each person who suffered). Financial impact will be massive. Most of the flooding occured in areas known to be at risk of flooding, though extending further than usual. There were a lot of people who were worried that they might be flooded during the worst of the rain, but wound up ok. We had almost exactly 1 m (yes, meter) of rain at our house, and the worst the flooding got was 6-8" in the street. Houston is well designed for floods!
> 
> 
> > On Aug 31, 2017, at 9:34 AM, Markus Stabrin <notifications@github.com> wrote:
> > 
> > Hello Steve,
> > 
> > Glad to hear that things are recovering well and that you all are fine and thank you for the status update.
> > 
> > As the news all over the world are full of Houston updates (and the pictures are still horrible) I, of course, did not expect a fast answer.
> > 
> > Best wishes to all of you!
> > Markus
> > 
> > On 31 Aug 2017, at 16:21, Steve Ludtke <notifications@github.com<mailto:notifications@github.com>> wrote:
> > 
> > Hi Markus,
> > as you may be aware we have been going through a significant weather event the last few days. Tunay is fine, but he was fairly close to being flooded, and none of us got much done for the last 5 days or so. I expect he will be coming back online shortly. Things are recovering well now. Please be patient.
> > 
> > > On Aug 31, 2017, at 8:38 AM, Markus Stabrin <notifications@github.com<mailto:notifications@github.com>> wrote:
> > >
> > > We figured out, that the pydusa recipe is always using the latest master branch and this resulted in a broken installer a couple of days ago.
> > >
> > > I added version tags to the pydusa repository and changed the recipe/installation script, so that users will use the default version (in this case tag v20170831), but developers do still have the freedom to build other tags/branches by adding a second argument next to the numpy version.
> > >
> > > @shadowwalkersb <https://github.com/shadowwalkersb> Do you agree with the changes or do you have another way of doing it?
> > > I am not really familiar with anaconda recipes yet.
> > >
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/cryoem/eman2/pull/119 <https://github.com/cryoem/eman2/pull/119>
> > > Commit Summary
> > >
> > > Changed pydusa recipe to make other versions available
> > > Removed source activate root again
> > > File Changes
> > >
> > > M recipes/pydusa/meta.yaml <https://github.com/cryoem/eman2/pull/119/files#diff-0> (4)
> > > M utils/build_pydusa_numpy.sh <https://github.com/cryoem/eman2/pull/119/files#diff-1> (9)
> > > Patch Links:
> > >
> > > https://github.com/cryoem/eman2/pull/119.patch <https://github.com/cryoem/eman2/pull/119.patch>
> > > https://github.com/cryoem/eman2/pull/119.diff <https://github.com/cryoem/eman2/pull/119.diff>
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6vQsxJ5iIdUMo-hxNx1mz-kGlOtTks5sdrdfgaJpZM4PI14s>.
> > >
> > 
> > ----------------------------------------------------------------------------
> > Steven Ludtke, Ph.D.
> > Professor, Dept of Biochemistry and Mol. Biol. (www.bcm.edu/biochem<http://www.bcm.edu/biochem>)
> > Co-Director National Center For Macromolecular Imaging (ncmi.bcm.edu<http://ncmi.bcm.edu>)
> > Co-Director CIBR Center (www.bcm.edu/research/cibr<http://www.bcm.edu/research/cibr>)
> > Baylor College of Medicine
> > sludtke@bcm.edu<mailto:sludtke@bcm.edu>
> > 
> > 
> > 
> > 
> > 
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub<https://github.com/cryoem/eman2/pull/119#issuecomment-326310676>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ARNjSBsXuOpr5pG2cOydc9Ku64IUR-Sgks5sdsFrgaJpZM4PI14s>.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub <https://github.com/cryoem/eman2/pull/119#issuecomment-326314790>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJ6i6l4SMEXxUZqdzbyA6Fgqu6jtnib_ks5sdsR3gaJpZM4PI14s>.
> > 
> 
> ----------------------------------------------------------------------------
> Steven Ludtke, Ph.D.
> Professor, Dept of Biochemistry and Mol. Biol. (www.bcm.edu/biochem)
> Co-Director National Center For Macromolecular Imaging (ncmi.bcm.edu)
> Co-Director CIBR Center (www.bcm.edu/research/cibr)
> Baylor College of Medicine 
> sludtke@bcm.edu
> 
> 
> 
> 
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

shadowwalkersb commented:
@mstabrin , if master on https://github.com/cryoem/pydusa is broken, why not just fix it?

shadowwalkersb commented on recipes/pydusa/meta.yaml:
```diff
@@ -3,8 +3,8 @@ package:
 source:
-    url: https://github.com/cryoem/pydusa/archive/master.tar.gz
-    fn: pydusa-master.tar.gz
+    url: https://github.com/cryoem/pydusa/archive/{{environ.get('GIT_PYDUSA_VERSION', 'v20170831')}}.tar.gz
```
`GIT_PYDUSA_VERSION` needs to be set in `build_pydusa_numpy.sh` before `conda-build` is called and needs to be whitelisted in this file, `meta.yaml`.

shadowwalkersb commented on utils/build_pydusa_numpy.sh:
```diff
@@ -17,12 +17,17 @@ set -xe
 RECIPES_DIR=$(cd $(dirname $0)/../recipes && pwd -P)
 
-if [ ${#args} -lt 1 ];then
+number_of_words=$(wc -w <<< "${args[@]}")
```
If this is to get number of arguments, `${#args}` gives exactly that without the need for a subshell, `$(...)`.

shadowwalkersb commented on utils/build_pydusa_numpy.sh:
```diff
@@ -17,12 +17,17 @@ set -xe
 
-if [ ${#args} -lt 1 ];then
+number_of_words=$(wc -w <<< "${args[@]}")
+if [ ${number_of_words} -le 0 ];then
```
If we are going to add more arguments/options, I would suggest replacing if/else with case statements and adding a `print_usage` function like it's done here, https://github.com/cryoem/build-scripts/blob/master/cronjob.sh.

mstabrin commented:
Changes reflected in PR #121
